framework/src/modules/interoperability/mainchain/commands/recover_message.ts

```diff
@@ -113,6 +113,18 @@ export class RecoverMessageCommand extends BaseInteroperabilityCommand<Mainchain
 			};
 		}
 
+		// Check that the CCM indices do not exceed the outbox size. We check only the last one,
+		// as the idxs are sorted in ascending order.
+		if (
+			terminatedOutboxAccount.outboxSize <=
+			idxs[idxs.length - 1] - 2 ** terminatedOutboxAccount.outboxSize
```

Comment:
I assume that `2 ** terminatedOutboxAccount.outboxSize` is meant to be the prepended `1` to the binary representation of the index? If yes, then it must be `2**height(terminatedChainOutbox)` ... but of course we don't have `height(terminatedChainOutbox)`. Maybe it's easier to remove the most significant bit from `idxs[idxs.length - 1]` instead.